### ISSUE TITLE
Error `owner.lookup is not a function` after upgrading to Ember 3.20.2

### ISSUE THREAD

**Issue by csprocket777** [onechiporenko/ember-models-table#426]
* **I'm submitting a ...**

  - [X] bug report
  - [ ] feature request
  - ~[ ] support request~ Ask in the [official Ember Discord server](https://discord.gg/zT3asNS).

* **What is the current behavior?**
A console error at bootup.
```
default.js:15 Uncaught (in promise) TypeError: owner.lookup is not a function
    at Class.<anonymous> (default.js:15)
    at index.js:2135
    at untrack (validator.js:780)
    at ComputedProperty.get (index.js:2134)
    at Class.CPGETTER_FUNCTION [as tableComponent] (index.js:1124)
    at getPossibleMandatoryProxyValue (index.js:1380)
    at get (index.js:1446)
    at reference.js:657
    at runInAutotrackingTransaction (validator.js:120)
    at track (validator.js:760)
```

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

* **[Bug] What is the expected behavior?**
No console error

* **[Feature] What is the motivation / use case for changing the behavior?**

* **[Support] What exactly you try to do?**
Upgrade my app from Ember `3.12` to `3.20`.

* **Please tell us about your environment:**

- `ember-models-table` ^3.3.0: 
- `ember` 3.20.2: 
- `ember-data` 3.20.0: 
- `ember-cli` 3.20.0:  
- `node` v10.20.1:
- `npm` or `yarn` version: npm 6.14.4, yarn 1.22.4

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

Stack Trace:
```

(anonymous) | @ | default.js:15
-- | -- | --
  | (anonymous) | @ | index.js:2135
  | untrack | @ | validator.js:780
  | get | @ | index.js:2134
  | CPGETTER_FUNCTION | @ | index.js:1124
  | getPossibleMandatoryProxyValue | @ | index.js:1380
  | get | @ | index.js:1446
  | (anonymous) | @ | reference.js:657
  | runInAutotrackingTransaction | @ | validator.js:120
  | track | @ | validator.js:760
  | value | @ | reference.js:656
  | value | @ | runtime.js:2585
  | initialize | @ | reference.js:82
  | peek | @ | reference.js:51
  | (anonymous) | @ | runtime.js:2962
  | evaluate | @ | runtime.js:2192
  | evaluateSyscall | @ | runtime.js:5111
  | evaluateInner | @ | runtime.js:5067
  | evaluateOuter | @ | runtime.js:5059
  | next | @ | runtime.js:5995
  | execute | @ | runtime.js:5970
  | handleException | @ | runtime.js:5237
  | handleException | @ | runtime.js:5414
  | throw | @ | runtime.js:5158
  | evaluate | @ | runtime.js:3020
  | execute | @ | runtime.js:5141
  | rerender | @ | runtime.js:5442
  | render | @ | index.js:8779
  | runInAutotrackingTransaction | @ | validator.js:120
  | (anonymous) | @ | index.js:9072
  | inTransaction | @ | runtime.js:2147
  | _renderRoots | @ | index.js:9046
  | _renderRootsTransaction | @ | index.js:9107
  | _revalidate | @ | index.js:9149
  | invoke | @ | backburner.js:338
  | flush | @ | backburner.js:229
  | flush | @ | backburner.js:426
  | _end | @ | backburner.js:960
  | Backburner._boundAutorunEnd | @ | backburner.js:629
  | Promise.then (async) |   |  
  | iterations | @ | backburner.js:28
  | flush | @ | index.js:43
  | _scheduleAutorun | @ | backburner.js:1179
  | _end | @ | backburner.js:970
  | Backburner._boundAutorunEnd | @ | backburner.js:629
  | Promise.then (async) |   |  
  | iterations | @ | backburner.js:28
  | flush | @ | index.js:43
  | _scheduleAutorun | @ | backburner.js:1179
  | _end | @ | backburner.js:970
  | Backburner._boundAutorunEnd | @ | backburner.js:629
  | Promise.then (async) |   |  
  | iterations | @ | backburner.js:28
  | flush | @ | index.js:43
  | _scheduleAutorun | @ | backburner.js:1179
  | _ensureInstance | @ | backburner.js:1167
  | schedule | @ | backburner.js:776
  | schedule | @ | index.js:361
  | (anonymous) | @ | index.js:524
  | (anonymous) | @ | index.js:512
  | flushAsyncObservers | @ | index.js:510
  | onEnd | @ | index.js:35
  | _end | @ | backburner.js:981
  | end | @ | backburner.js:710
  | _run | @ | backburner.js:1015
  | _join | @ | backburner.js:989
  | join | @ | backburner.js:760
  | join | @ | index.js:168
  | (anonymous) | @ | ajax-request.js:256
  | fire | @ | jquery.js:3496
  | fireWith | @ | jquery.js:3626
  | done | @ | jquery.js:9786
  | (anonymous) | @ | jquery.js:10047
  | load (async) |   |  
  | send | @ | jquery.js:10066
  | ajax | @ | jquery.js:9682
  | _makeRequest | @ | ajax-request.js:236
  | request | @ | ajax-request.js:195
  | (anonymous) | @ | data-management.js:1822
  | (anonymous) | @ | index.js:2135
  | untrack | @ | validator.js:780
  | get | @ | index.js:2134
  | CPGETTER_FUNCTION | @ | index.js:1124
  | getPossibleMandatoryProxyValue | @ | index.js:1380
  | get | @ | index.js:1446
  | _getPath | @ | index.js:1495
  | get | @ | index.js:1437
  | get | @ | observable.js:116
  | (anonymous) | @ | data-management.js:1910
  | (anonymous) | @ | index.js:2135
  | untrack | @ | validator.js:780
  | get | @ | index.js:2134
  | CPGETTER_FUNCTION | @ | index.js:1124
  | getPossibleMandatoryProxyValue | @ | index.js:1380
  | get | @ | index.js:1446
  | (anonymous) | @ | reference.js:657
  | runInAutotrackingTransaction | @ | validator.js:120
  | track | @ | validator.js:760
  | value | @ | reference.js:656
  | value | @ | runtime.js:4818
  | processComponentArgs | @ | index.js:3737
  | create | @ | index.js:3958
  | (anonymous) | @ | runtime.js:3559
  | evaluate | @ | runtime.js:2192
  | evaluateSyscall | @ | runtime.js:5111
  | evaluateInner | @ | runtime.js:5067
  | evaluateOuter | @ | runtime.js:5059
  | next | @ | runtime.js:5995
  | execute | @ | runtime.js:5970
  | handleException | @ | runtime.js:5237
  | handleException | @ | runtime.js:5414
  | throw | @ | runtime.js:5158
  | evaluate | @ | runtime.js:3020
  | execute | @ | runtime.js:5141
  | rerender | @ | runtime.js:5442
  | render | @ | index.js:8779
  | runInAutotrackingTransaction | @ | validator.js:120
  | (anonymous) | @ | index.js:9072
  | inTransaction | @ | runtime.js:2147
  | _renderRoots | @ | index.js:9046
  | _renderRootsTransaction | @ | index.js:9107
  | _revalidate | @ | index.js:9149
  | invoke | @ | backburner.js:338
  | flush | @ | backburner.js:229
  | flush | @ | backburner.js:426
  | _end | @ | backburner.js:960
  | Backburner._boundAutorunEnd | @ | backburner.js:629
  | Promise.then (async) |   |  
  | iterations | @ | backburner.js:28
  | flush | @ | index.js:43
  | _scheduleAutorun | @ | backburner.js:1179
  | _end | @ | backburner.js:970
  | Backburner._boundAutorunEnd | @ | backburner.js:629
  | Promise.then (async) |   |  
  | iterations | @ | backburner.js:28
  | flush | @ | index.js:43
  | _scheduleAutorun | @ | backburner.js:1179
  | _end | @ | backburner.js:970
  | Backburner._boundAutorunEnd | @ | backburner.js:629
  | Promise.then (async) |   |  
  | iterations | @ | backburner.js:28
  | flush | @ | index.js:43
  | _scheduleAutorun | @ | backburner.js:1179
  | _end | @ | backburner.js:970
  | Backburner._boundAutorunEnd | @ | backburner.js:629
  | Promise.then (async) |   |  
  | iterations | @ | backburner.js:28
  | flush | @ | index.js:43
  | _scheduleAutorun | @ | backburner.js:1179
  | _end | @ | backburner.js:970
  | Backburner._boundAutorunEnd | @ | backburner.js:629
  | Promise.then (async) |   |  
  | iterations | @ | backburner.js:28
  | flush | @ | index.js:43
  | _scheduleAutorun | @ | backburner.js:1179
  | _ensureInstance | @ | backburner.js:1167
  | schedule | @ | backburner.js:776
  | (anonymous) | @ | rsvp.js:19
  | fulfill | @ | rsvp.js:428
  | handleMaybeThenable | @ | rsvp.js:384
  | resolve$1 | @ | rsvp.js:401
  | resolved | @ | rsvp.js:526
  | success | @ | data-management.js:1758
  | fire | @ | jquery.js:3496
  | fireWith | @ | jquery.js:3626
  | done | @ | jquery.js:9786
  | (anonymous) | @ | jquery.js:10047
  | load (async) |   |  
  | send | @ | jquery.js:10066
  | ajax | @ | jquery.js:9682
  | _makeRequest | @ | ajax-request.js:236
  | request | @ | ajax-request.js:195
  | _emberData.default.PromiseObject.create.promise | @ | data-management.js:1752
  | initializePromise | @ | rsvp.js:520
  | Promise | @ | rsvp.js:1021
  | requestDashboardLayout | @ | data-management.js:1751
  | bootForConsumptionPostGraphChoice | @ | app-boot-pipeline.js:431
  | async function (async) |   |  
  | bootForConsumptionPostGraphChoice | @ | app-boot-pipeline.js:415
  | consumeBoot | @ | app-boot-pipeline.js:210
  | routeBoot | @ | app-boot-pipeline.js:135
  | boot | @ | app-boot-pipeline.js:115
  | async function (async) |   |  
  | boot | @ | app-boot-pipeline.js:90
  | didTransition | @ | index.js:34
  | triggerEvent | @ | router.js:1296
  | triggerEvent | @ | router.js:180
  | finalizeTransition | @ | router_js.js:1615
  | (anonymous) | @ | router_js.js:1542
  | invokeCallback | @ | rsvp.js:493
  | publish | @ | rsvp.js:476
  | (anonymous) | @ | rsvp.js:19
  | invoke | @ | backburner.js:338
  | flush | @ | backburner.js:229
  | flush | @ | backburner.js:426
  | _end | @ | backburner.js:960
  | end | @ | backburner.js:710
  | _run | @ | backburner.js:1015
  | _join | @ | backburner.js:989
  | join | @ | backburner.js:760
  | join | @ | index.js:168
  | (anonymous) | @ | index.js:263
  | mightThrow | @ | jquery.js:3762
  | process | @ | jquery.js:3830
  | setTimeout (async) |   |  
  | (anonymous) | @ | jquery.js:3868
  | fire | @ | jquery.js:3496
  | fireWith | @ | jquery.js:3626
  | fire | @ | jquery.js:3634
  | fire | @ | jquery.js:3496
  | fireWith | @ | jquery.js:3626
  | ready | @ | jquery.js:4106
  | completed | @ | jquery.js:4116


```

**Comment by onechiporenko** [comment]
Hello, @csprocket777. Thanks for report! Could  you please provide some demo-app with reproduction? I've tried to reproduce this issue and couldn't do this.

**Comment by csprocket777** [comment]
We are trying to narrow down the problem and suspect it has to do with how we're implementing a customized theme for the addon. We had it defined in a service and were bringing it in that way and are moving to do the initializer route. Basically, the reference to `this` isn't correct and is causing the `owner.lookup` to fail.

**Comment by kallenboone** [comment]
I fixed the issue. The problem was that the `@componentPath` decorator in the `Default` theme required the theme to be injected into a component, or else the `owner.lookup` would fail. I fixed it by creating an initializer to register our custom theme, then injecting it into components using an instance initializer, instead of passing the custom theme as an argument to the `ModelTable` components. 

The documentation in GitHub explains how to do the injection of existing themes, but it doesn't cover how to create custom themes, register them in an initializer, and inject them with an instance initializer. It would be helpful if these steps were included in the README.
